test/storage.js

```diff
@@ -0,0 +1,64 @@
+var lastTest = "",
+	done = false,
+	mockStorage = {
+		_store: Object.create( null ),
+		setItem: function( key, value ) {
+			this._store[ key ] = value;
+			this.length++;
+		},
+		getItem: function( key ) {
+			return key in this._store ? this._store[ key ] : null;
+		},
+		removeItem: function( key ) {
+			if ( key in this._store ) {
+				delete this._store[ key ];
+				this.length--;
+			}
+		},
+		key: function( i ) {
+			return Object.keys( this._store )[ i ];
+		},
+		length: 0
+	};
+
+mockStorage.setItem( "qunit-test-Storage-Should be run first", 1 );
+QUnit.config.storage = mockStorage;
+
+QUnit.module( "Storage", function() {
+	QUnit.test( "Passing test", function( assert ) {
+		assert.strictEqual( lastTest, "Should be run first" );
+		mockStorage.setItem( "qunit-test-Storage-Passing test", true );
+		assert.ok( true );
+	} );
+
+	QUnit.test( "Removes passing tests from storage", function( assert ) {
+		assert.strictEqual( mockStorage.getItem( "qunit-test-Storage-Passing test" ), null );
+	} );
+
+	// Verifies test reordering by storage values
+	QUnit.test( "Should be run first", function( assert ) {
+		assert.strictEqual( lastTest, "" );
+		lastTest = "Should be run first";
+	} );
+} );
+
+QUnit.done( function() {
+	if ( done ) {
+		return;
+	}
+
+	done = true;
+
+	mockStorage.setItem( "qunit-test-should-remove", true );
+	mockStorage.setItem( "should-not-remove", true );
+	mockStorage.setItem( "qunit-test-should-also-remove", true );
+
+	// TODO: Is there a way to test this that will show up in CI?
```

Comment:
It would, and that may be more trouble than it's worth. I guess we should make changes here to handle multiple `done` in Node, and then just skip those assertions in phantomjs.